apis/v1beta1/httproute_types.go

```diff
@@ -205,8 +205,10 @@ type HTTPRouteRule struct {
 	// All filters are expected to be compatible with each other except for the
 	// URLRewrite and RequestRedirect filters, which may not be combined. If an
 	// implementation can not support other combinations of filters, they must clearly
-	// document that limitation. In all cases where incompatible or unsupported
-	// filters are specified, implementations MUST add a warning condition to status.
+	// document that limitation. In cases where incompatible or unsupported
+	// filters are specified on subset of the rules on a route, implementations
+	// MUST add the `PartiallyInvalid` error condition with status `True` and
+	// reason `IncompatibleFilters`.
```

Comment:
This PR feels like it's combining 2 improvements that are related but different:

1. A new `PartiallyInvalid` reason when only a subset or rules in a Route are invalid
2. A new `IncompatibleFilters` reason

It seems like the following reasons could apply to an individual rule or the entire route:

* IncompatibleFilters
* UnsupportedValue (Accepted)
* BackendNotFound (ResolvedRefs)
* InvalidKind (ResolvedRefs)
* RefNotPermitted (ResolvedRefs)

Currently `ResolvedRefs` is only true if 100% of refs are resolved, that's _probably_ fine to leave as is, but it may be confusing for `ResolvedRefs` to be true and `PartiallyInvalid` to be false.

I think the simplest path forward would be to add `IncompatibleFilters` as a possible reason for `Accepted`. Adding a `PartiallyInvalid` condition to reflect that a portion of rules are invalid makes sense, but I think we'd need to make broader changes to the other conditions on the route to have that make sense.

Another option would be to say that both `IncompatibleFilters` and `UnsupportedValue` reasons could be used for `Accepted` and `PartiallyInvalid` conditions. When used with the `Accepted` condition, nothing in the Route would be reconciled, but when used with the `PartiallyInvalid` condition, the valid subset of the Route would be reconciled. We could potentially leave `ResolvedRefs` and the associated reasons as a special case that always applies to the entire route since I think our recommended behavior is just to return 404s in those cases and not actually to stop reconciling anything. I'm less sure about this approach though, so maybe it's better as a follow up.

Reply:
added https://github.com/kubernetes-sigs/gateway-api/pull/2150 to start over and not have to disentagle the changes from this PR